cfme/tests/containers/test_container_provider_crud.py

```diff
@@ -0,0 +1,71 @@
+import fauxfactory
+import pytest
+
+from cfme.common.provider_views import ContainerProvidersView
+from cfme.containers.provider import ContainersProvider
+from cfme.fixtures.provider import setup_or_skip
+from cfme.utils.appliance.implementations.ui import navigate_to
+
+
+pytestmark = [
+    pytest.mark.tier(1),
+    pytest.mark.provider([ContainersProvider], scope='function')
+]
+
+
+@pytest.mark.tier(3)
+def test_container_provider_crud(request, appliance, provider):
+
+    """
+    Polarion:
+        assignee: juwatts
+        caseimportance: medium
+        casecomponent: Containers
+        initialEstimate: 1/6h
+    """
+
+    if provider.exists:
```

Comment:
I think instead of this pattern, you can use the `has_no_providers` fixture to clear them out before the test starts.